### PR TITLE
Add custom repo functionality

### PR DESCRIPTION
**Pull request by matoval**
##### SUMMARY

Add custom repo functionality

Add ability to add custom repos as a new source for custom packages

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by maxamillion** [comment]
I would propose that we rename `plugins/modules/add_source.py` to `plugins/modules/repository.py` and add a parameter called `state` that allows either `present` or `absent` similar to the `package`/`yum`/`dnf` modules, then add the logic to handle creation, deletion, and change detection for the repository.

**Review by maxamillion — CHANGES_REQUESTED** [review]
I would propose that we rename `plugins/modules/add_source.py` to `plugins/modules/repository.py` and add a parameter called state that allows either present or absent similar to the `package`/`yum`/`dnf` modules, then add the logic to handle creation, deletion, and change detection for the repository.

File: roles/setup_server/tasks/main.yaml [hunk 1]
```diff
@@ -15,6 +15,16 @@
     - '"rh-amazon-rhui-client" not in ansible_facts["packages"]'
     - '"rhui-azure-rhel9" not in ansible_facts["packages"]'
     - '"rhui-azure-rhel8" not in ansible_facts["packages"]'
+
```
[the remark this answers is not in the document]
Comment (chadmf) [reply roles/setup_server/tasks/main.yaml:18]: Fixed with last push thanks!

**Review by chadmf — CHANGES_REQUESTED** [review]
Code works, however it needs to be added to the README.md on how to use this functionality.

**Review by maxamillion — APPROVED** [review]

**Review by maxamillion — CHANGES_REQUESTED** [review]
Needs integration tests

**Comment by chadmf** [comment]
LGTM!

**Comment by maxamillion** [comment]
merging early because it's passed integration tests during our debug session on matrix and I need to make a PR with a lot of changes to make ansible-lint happy

**Pull request merged**